src/displayapp/screens/WatchFacePineTimeStyle.cpp

```diff
@@ -527,34 +520,36 @@ void WatchFacePineTimeStyle::Refresh() {
     }
   }
 
-  stepCount = motionController.NbSteps();
-  if (stepCount.IsUpdated()) {
-    lv_gauge_set_value(stepGauge, 0, (stepCount.Get() / (settingsController.GetStepsGoal() / 100)) % 100);
-    lv_obj_realign(stepGauge);
-    lv_label_set_text_fmt(stepValue, "%luK", (stepCount.Get() / 1000));
-    lv_obj_realign(stepValue);
-    if (stepCount.Get() > settingsController.GetStepsGoal()) {
-      lv_obj_set_style_local_line_color(stepGauge, LV_GAUGE_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
-      lv_obj_set_style_local_scale_grad_color(stepGauge, LV_GAUGE_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+  if (settingsController.IsWidgetOn(Pinetime::Controllers::Settings::Widget::Steps)) {
+    stepCount = motionController.NbSteps();
+    if (stepCount.IsUpdated()) {
+      lv_gauge_set_value(stepGauge, 0, (stepCount.Get() / (settingsController.GetStepsGoal() / 100)) % 100);
+      lv_obj_realign(stepGauge);
+      lv_label_set_text_fmt(stepValue, "%luK", (stepCount.Get() / 1000));
+      lv_obj_realign(stepValue);
+      if (stepCount.Get() > settingsController.GetStepsGoal()) {
+        lv_obj_set_style_local_line_color(stepGauge, LV_GAUGE_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+        lv_obj_set_style_local_scale_grad_color(stepGauge, LV_GAUGE_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+      }
     }
   }
 
-  currentWeather = weatherService.Current();
-  if (currentWeather.IsUpdated()) {
-    auto optCurrentWeather = currentWeather.Get();
-    if (optCurrentWeather) {
-      int16_t temp = optCurrentWeather->temperature.Celsius();
-      if (settingsController.GetWeatherFormat() == Controllers::Settings::WeatherFormat::Imperial) {
-        temp = optCurrentWeather->temperature.Fahrenheit();
+  if (settingsController.IsWidgetOn(Pinetime::Controllers::Settings::Widget::Weather)) {
+    currentWeather = weatherService.Current();
+    if (currentWeather.IsUpdated()) {
+      auto optCurrentWeather = currentWeather.Get();
+      if (optCurrentWeather) {
+        int16_t temp = optCurrentWeather->temperature.Celsius();
+        if (settingsController.GetWeatherFormat() == Controllers::Settings::WeatherFormat::Imperial) {
+          temp = optCurrentWeather->temperature.Fahrenheit();
+        }
+        lv_label_set_text_fmt(temperature, "%d°", temp);
+        lv_label_set_text(weatherIcon, Symbols::GetSymbol(optCurrentWeather->iconId));
+      } else {
+        lv_label_set_text(temperature, "--");
+        lv_label_set_text(weatherIcon, Symbols::ban);
       }
-      lv_label_set_text_fmt(temperature, "%d°", temp);
-      lv_label_set_text(weatherIcon, Symbols::GetSymbol(optCurrentWeather->iconId));
-    } else {
-      lv_label_set_text(temperature, "--");
-      lv_label_set_text(weatherIcon, Symbols::ban);
     }
-    lv_obj_realign(temperature);
```

Comment:
I'm not great at LVGL, but just want to check that removing these realigns is intentional